src/library/Button/Button.js

```diff
@@ -291,7 +291,7 @@ function filterProps({ element, type }: Props) {
   // these are not filtered automatically by rootEl
   const invalidComponentProps = ['primary', 'text', 'variant'];
   const invalidLinkProps =
-    element === 'a' && isTypeButton(type) ? ['type'] : [];
+    element === 'a' && isTypeButton(type) ? ['type'] : ['element'];
```

Comment:
I noticed that the `type` attribute is still being passed to the dom, even when the ultimately rendered HTML element is an `a`, which is invalid. I wonder if there's a way to prevent this. Maybe check for `href` or `to` props? Or only apply if an `onClick` is provided?